src/hash.ts

```diff
@@ -19,9 +19,23 @@ export const keccak = function(a: any, bits: number = 256): Buffer {
 
   if (!bits) bits = 256
 
-  return createKeccakHash(`keccak${bits}`)
-    .update(a)
-    .digest()
+  switch (bits) {
+    case 224: {
+      return keccak224(a)
+    }
+    case 256: {
+      return k256(a)
+    }
+    case 384: {
+      return keccak384(a)
+    }
+    case 512: {
+      return keccak512(a)
+    }
+    default: {
+      throw new Error(`Invald algorithm: keccak${bits}`)
```

Comment:
Note for reviewers: There aren't versions of keccak other than 224, 256, 384, and 512. This API is confusing, but we have to keep it for backward compatibility reasons.